app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,27 @@
+class SessionsController < ApplicationController
+    get '/sessions/new' do
+      email = session[:sign_in_email] || ""
+      session[:sign_in_email] = nil
```

Comment:
Rather than storing the email they put in the form in the `session`, you can pass it into the `erb` template via. `locals`